### DISCUSSION TITLE
Does HxMessenger work in InteractiveServer render mode?

### DISCUSSION THREAD

**Discussion by joshberry** [havit/Havit.Blazor#783] [Q&A] [answered]
Following the instructions for setting up the HxMessenger here: https://havit.blazor.eu/components/HxMessenger, it simply doesn't work for me. I tried creating a new empty project using the default template and still nothing happens when calling Messenger.AddInformation() or any of the other Add* methods.

I have found that if I move the HxMessenger component from the layout component to the page component I'm running the code on, it does work. But then the messages are lost/not shown when navigating between pages. 

Home.razor

```
@page "/"
@rendermode InteractiveServer

<PageTitle>Home</PageTitle>

<h1>Hello, world!</h1>

Welcome to your new app.

<HxButton Text="Show information" OnClick="HandleAddInformationButtonClick" Color="ThemeColor.Primary" />

@code {
	[Inject] protected IHxMessengerService Messenger { get; set; }

	private void HandleAddInformationButtonClick()
	{
		Messenger.AddInformation(title: "Data encrypted", message: "All your data has been encrypted. To decrypt your data share $100 with all your teammates.");
	}
}

```

MainLayout.razor
```
@inherits LayoutComponentBase
<HxMessenger Position="ToastContainerPosition.TopEnd" />
<div class="page">
    <div class="sidebar">
        <NavMenu />
    </div>

    <main>
        <div class="top-row px-4">
            <a href="https://learn.microsoft.com/aspnet/core/" target="_blank">About</a>
        </div>

        <article class="content px-4">
            @Body
        </article>
    </main>
</div>

<div id="blazor-error-ui">
    An unhandled error has occurred.
    <a href="" class="reload">Reload</a>
    <a class="dismiss">🗙</a>
</div>
```

Am I doing something wrong, or is the HxMessenger and service not functional when using InteractiveServer rendermode?

Thanks for any help!

**Comment by hakenr** [comment, marked as answer]
Where do you set the interactivity of your components? It seems you opted for "per page/component" which means your `MainLayout.razor` is non-interactive (and non of the interactive components will work there without further setting).

Use `<HxMessenger @rendermode="@InteractiveServer" ... />` when you use it in non-interactive `MainLayout.razor`.

**Reply by joshberry** [reply]
That was it. Thank you so much!
